### PR TITLE
feat(init): #MA-997 add initialization for 1d and 2d (incidents)

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
L'objectif du ticket MA-997 est de séparé le bouton d'initialisation de presence en 2 bouton d'initialisation, 1 pour les droits 1d, l'autre pour les droit 2d. C'est 2 bouton sont lier a un droit et cela va permettre avoir des données d'initialisation différent entre les établissements 1d et 2d

Doc: https://confluence.support-ent.fr/display/MP/Incident+API

## Checklist tests
Initialiser un établisement 1d
Initialiser un établisement 2d

## Issue ticket number and link
[MA-997](https://jira.support-ent.fr/browse/MA-997)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: incidents/src/main/java/fr/openent/incidents/helper/init/IInitIncidentsHelper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import fr.openent.presences.enums.InitTypeEnum;
+
+import java.util.List;
+
+public interface IInitIncidentsHelper {
+    Init1DIncidentsHelper init1DIncidentsHelper = new Init1DIncidentsHelper();
+    Init2DIncidentsHelper init2DIncidentsHelper = new Init2DIncidentsHelper();
+
+    static IInitIncidentsHelper getInstance(InitTypeEnum initTypeEnum) {
+        if (initTypeEnum == InitTypeEnum.ONE_D) {
```
Comment (flmartineau) [inline incidents/src/main/java/fr/openent/incidents/helper/init/IInitIncidentsHelper.java]: `return (initTypeEnum == InitTypeEnum.ONE_D) ? init1DIncidentsHelper : init2DIncidentsHelper`

File: incidents/src/main/java/fr/openent/incidents/helper/init/Init1DIncidentsHelper.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    public List<IncidentType> getIncidentTypes() {
+        return IInitIncidentsHelper.init2DIncidentsHelper.getIncidentTypes();
+    }
+
+    @Override
+    public List<Place> getPlaces() {
+        List<Place> placeList = new ArrayList<>();
+        for (int i = 0; i < 6; i++) {
```
Comment (flmartineau) [inline incidents/src/main/java/fr/openent/incidents/helper/init/Init1DIncidentsHelper.java]: Dans une variable

File: incidents/src/main/java/fr/openent/incidents/helper/init/Init1DIncidentsHelper.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    public List<Seriousness> getSeriousnessTypes() {
+        return IInitIncidentsHelper.init2DIncidentsHelper.getSeriousnessTypes();
+    }
+
+    @Override
+    public List<Partner> getPartners() {
+        List<Partner> partnerList = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
```
Comment (flmartineau) [inline incidents/src/main/java/fr/openent/incidents/helper/init/Init1DIncidentsHelper.java]: variable

File: incidents/src/main/java/fr/openent/incidents/helper/init/Init1DIncidentsHelper.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            partnerList.add(partner);
+        }
+        return partnerList;
+    }
+
+    @Override
+    public List<PunishmentType> getPunishmentTypes() {
+        List<PunishmentType> punishmentTypeList = new ArrayList<>();
+
+        punishmentTypeList.add(new PunishmentType(null, "Devoir supplémentaire", "PUNISHMENT", 1, false));
```
Comment (flmartineau) [inline incidents/src/main/java/fr/openent/incidents/helper/init/Init1DIncidentsHelper.java]: Pas possible de se baser sur les i18n ?

File: incidents/src/main/java/fr/openent/incidents/helper/init/Init2DIncidentsHelper.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import java.util.List;
+
+public class Init2DIncidentsHelper implements IInitIncidentsHelper {
+    protected Init2DIncidentsHelper() {
+    }
+
+    @Override
+    public List<IncidentType> getIncidentTypes() {
+        List<IncidentType> incidentTypeList = new ArrayList<>();
+        for (int i = 0; i < 6; i++) {
```
Comment (flmartineau) [inline incidents/src/main/java/fr/openent/incidents/helper/init/Init2DIncidentsHelper.java]: variable pour le 6 (pareil ensuite)

File: incidents/src/main/java/fr/openent/incidents/helper/init/Init2DIncidentsHelper.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            partnerList.add(partner);
+        }
+        return partnerList;
+    }
+
+    @Override
+    public List<PunishmentType> getPunishmentTypes() {
+        List<PunishmentType> punishmentTypeList = new ArrayList<>();
+
+        punishmentTypeList.add(new PunishmentType(null, "Devoir supplémentaire", "PUNISHMENT", 1, false));
```
Comment (flmartineau) [inline incidents/src/main/java/fr/openent/incidents/helper/init/Init2DIncidentsHelper.java]: Possible d'utiliser les i18n ?

**Review by adrienFro — APPROVED** [review]

**Review by flmartineau — APPROVED** [review]

**Pull request merged**